include/aie/Dialect/AIE/IR/AIE.td

```diff
@@ -1117,19 +1117,12 @@ def AIE_LockOp: AIE_Op<"lock", [TileElement]>, Results<(outs Index)> {
 
   let extraClassDeclaration = [{
     bool hasName() {
-      if(auto attr = getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName())) {
-        return true;
-      } else {
-        return false;
-      }
+      return bool(getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName()));
     }
     StringAttr name() {
-      if(auto attr = getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName())) {
+      if(auto attr = getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName()))
         return attr;
-      } else {
-        emitOpError("does not have '") << SymbolTable::getSymbolAttrName() <<
-          "' attribute specified";
-      }
+      emitOpError("does not have '") << SymbolTable::getSymbolAttrName() << "' attribute specified";
       llvm_unreachable("unreachable");
```

Comment:
This is clearly reachable.  The llvm_unreachable is bogus.

Reply:
Got me - there's a compile error without this.

Reply:
I think it should probably be an assert, not unreachable.

Reply:
Sorry I take my thumbs up back - how is that reachable if there's an `emitOpError`?

Reply:
I think emitOpError prints a message, but doesn't fail.  in most circumstances, it's part of a transform where you return an expected, but obviously that can't happen in this situation.

Reply:
Ah you're right - I was thinking of `report_fatal_error`.